micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/CompilationMetrics.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.jvm;
+
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.binder.MeterBinder;
+import io.micrometer.core.lang.NonNullApi;
+import io.micrometer.core.lang.NonNullFields;
+
+import java.lang.management.CompilationMXBean;
+import java.lang.management.ManagementFactory;
+
+import static java.util.Collections.emptyList;
+
+@NonNullApi
+@NonNullFields
+public class CompilationMetrics implements MeterBinder {
+    private final Iterable<Tag> tags;
+
+    public CompilationMetrics() {
+        this(emptyList());
+    }
+
+    public CompilationMetrics(Iterable<Tag> tags) {
+        this.tags = tags;
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        CompilationMXBean compilationBean = ManagementFactory.getCompilationMXBean();
+        if (compilationBean.isCompilationTimeMonitoringSupported()) {
+            FunctionCounter.builder("jvm.compilation.time.total", compilationBean, CompilationMXBean::getTotalCompilationTime)
```

Comment:
I am unsure about the naming and the baseUnit usage below.
The PrometheusNamingConvention adds the unit and then the "_total" suffix again due to how it works for `Counter`s:
`jvm_compilation_time_total_ms_total`

Unfortunately we can't make this a `Timer` due to not being triggered by the JVM so we don't really have any counts on compilation frequency.

Reply:
What about `jvm.compilation.ms.total` ?

Reply:
That will become `jvm_compilation_ms_total_ms_total` because the `PrometheusNamingConvention` will first add the base unit and then the `_total`.

How about just `jvm.compilation.time` and any NamingConvention specific to a registry implementation will make sure that it generates whatever is best practice for that backend. For Prometheus this would then become `jvm_compilation_time_ms_total`.

Reply:
`jvm.compilation.time` sounds good to me.